examples/mongodb-atlas/mongodb-atlas-blueprint.yaml

```diff
@@ -0,0 +1,193 @@
+            counter=40
+            while [[ $isCreated != true && $isFailed == false ]];
+            do
+              atlas backups snapshots describe {{ .Phases.mongoBackup.Output.snapshotId }}\
+              --clusterName ${cluster_name} -o json > output.json
+              isCompleted=$(jq -r ".status" output.json)
+              if [ $isCompleted == "failed" ]; then
+                exit 1
+              fi
+              if [ $isCompleted == "completed" ]; then
+                exit 0
+              fi
+              sleep 30
+              if [[ $counter -ge 40 ]]; then
+                echo "Timed out waiting for backup completion."
+                exit 1
+              fi
+              counter=$((counter+1))
+            done
+
+  restore:
+    inputArtifactNames:
+      - backupOutput
+    phases:
+    - func: KubeTask
+      name: mongoRestore
+      args:
+        namespace: "{{ .Object.metadata.namespace }}"
+        image: <registry>/<repository>/mongodb-atlas:<tag_name>
+        command:
+          - bash
+          - -o
+          - errexit
+          - -o
+          - pipefail
+          - -c
+          - |
+            cat >/root/.config/atlascli/config.toml <<EOL
+            [default]
+            org_id = '{{ .Object.data.orgid | toString | b64dec }}'
+            output = 'json'
+            private_api_key = '{{ .Object.data.privatekey | toString | b64dec }}'
+            project_id = '{{ .Object.data.projectid | toString | b64dec }}'
+            public_api_key = '{{ .Object.data.publickey | toString | b64dec }}'
+            service = 'cloud'
+            EOL
+            cluster_name='{{ .Object.data.clustername | toString | b64dec }}'
+            snapshot_id='{{ .ArtifactsIn.backupOutput.KeyValue.snapshotId }}'
+            target_cluster_name=${cluster_name}
+            target_project_id='{{ .Object.data.projectid | toString | b64dec }}'
+            atlas backup restore start automated --clusterName ${cluster_name} --snapshotId ${snapshot_id} \
+            --targetClusterName ${target_cluster_name} --targetProjectId ${target_project_id} -o json > output.json
+            restore_id=$(jq -r ".id" output.json)
+            kando output restoreId ${restore_id}
+
+    - func: KubeTask
+      name: waitForRestoreToComplete
+      args:
+        namespace: "{{ .Object.metadata.namespace }}"
+        image: <registry>/<repository>/mongodb-atlas:<tag_name>
+        command:
+          - bash
+          - -o
+          - errexit
+          - -o
+          - pipefail
+          - -c
+          - |
+            cat >/root/.config/atlascli/config.toml <<EOL
+            [default]
+            org_id = '{{ .Object.data.orgid | toString | b64dec }}'
+            output = 'json'
+            private_api_key = '{{ .Object.data.privatekey | toString | b64dec }}'
+            project_id = '{{ .Object.data.projectid | toString | b64dec }}'
+            public_api_key = '{{ .Object.data.publickey | toString | b64dec }}'
```

Comment:
@shahpratikr We do not pass objects during restore and delete actions.
We just refer to backup action to find required creds. We should store all the required fields as output artifact of backup action and refer to them during restore and delete